components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/AuthorizedAPIManagementServiceImpl.java

```diff
@@ -187,25 +190,42 @@ public void patchAuthorizedAPI(String appId, String apiId, List<String> addedSco
     public List<AuthorizedScopes> getAuthorizedScopes(String appId, String tenantDomain)
             throws IdentityApplicationManagementException {
 
-        Collection<AuthorizedAPIManagementListener> listeners = ApplicationMgtListenerServiceComponent
-                .getAuthorizedAPIManagementListeners();
-        for (AuthorizedAPIManagementListener listener : listeners) {
-            listener.preGetAuthorizedScopes(appId, tenantDomain);
-        }
-        // Check if the application is a main application else get the main application id and main tenant id.
-        ApplicationManagementService applicationManagementService = ApplicationManagementServiceImpl.getInstance();
-        String mainAppId = applicationManagementService.getMainAppId(appId);
-        if (mainAppId != null) {
-            appId = mainAppId;
-            int tenantId = applicationManagementService.getTenantIdByApp(mainAppId);
-            tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
-        }
-        List<AuthorizedScopes> authorizedScopes = authorizedAPIDAO.getAuthorizedScopes(appId,
-                IdentityTenantUtil.getTenantId(tenantDomain));
-        for (AuthorizedAPIManagementListener listener : listeners) {
-            listener.postGetAuthorizedScopes(authorizedScopes, appId, tenantDomain);
+        try {
+            Collection<AuthorizedAPIManagementListener> listeners = ApplicationMgtListenerServiceComponent
+                    .getAuthorizedAPIManagementListeners();
+            for (AuthorizedAPIManagementListener listener : listeners) {
+                listener.preGetAuthorizedScopes(appId, tenantDomain);
+            }
+            // Check if the application is a main application else get the main application id and main tenant id.
+            ApplicationManagementService applicationManagementService = ApplicationManagementServiceImpl.getInstance();
+            String mainAppId = applicationManagementService.getMainAppId(appId);
+            if (mainAppId != null) {
+                appId = mainAppId;
+                int tenantId = applicationManagementService.getTenantIdByApp(mainAppId);
+                tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+            }
+
+            List<AuthorizedScopes> authorizedScopes;
+            // If authorizeAllScopes is enabled, all scopes will be considered as authorized.
+            if (Boolean.parseBoolean(IdentityUtil.getProperty(AUTHORIZE_ALL_SCOPES))) {
```

Comment:
cant we use a separate function to parse bool  like `isAuthroizeAllScopesEnabled?`